### PR TITLE
Allocate contract - externals

### PR DESCRIPTION
**Pull request by MartinForre**

### CODE REVIEW

File: src/frontend/src/index.tsx [hunk 1]
```diff
@@ -8,9 +8,9 @@ import AppContext from './appContext';
 
 const App: React.FC = () => {
     const fusionContext = useFusionContext();
-    const apiClient = new ApiClient(
-        fusionContext.http.client,
-        'https://resources-api.ci.fusion-dev.net'
+    const apiClient = React.useMemo(
```
Comment (MartinForre) [inline src/frontend/src/index.tsx]: @eskil0312

File: src/frontend/src/hooks/useForm.ts [hunk 2]
```diff
@@ -1,43 +1,59 @@
-import { useState, useCallback, useMemo, useEffect } from "react";
+import { useState, useCallback, useMemo, useEffect } from 'react';
+const deepEqual = require('deep-equal');
```
Comment (eskil0312) [inline src/frontend/src/hooks/useForm.ts]: `import deepEqual from 'deep-equal'`?

Comment (MartinForre) [reply src/frontend/src/hooks/useForm.ts]: TypeScript wouldn't let me do that because of how the module is exported

Comment (eskil0312) [reply src/frontend/src/hooks/useForm.ts]: ahha

File: src/frontend/src/api/ApiClient.ts [hunk 3]
```diff
@@ -16,16 +23,109 @@ export default class ApiClient {
...
+        const response = await this.httpClient.putAsync<
+            Contract,
+            Contract,
+            FusionApiHttpErrorResponse
+        >(url, contract);
+
+        return response.data;
+    }
+
+    async createExternalCompanyReprasentiveAsync(
```
Comment (olerichard) [inline src/frontend/src/api/ApiClient.ts]: createExternalCompanyReprasentiveAsync should probably be createExternalCompanyRepresentativeAsync ?

Comment (MartinForre) [reply src/frontend/src/api/ApiClient.ts]: @olerichard fixed

File: src/frontend/src/api/ApiClient.ts [hunk 4]
```diff
@@ -17,22 +24,112 @@ export default class ApiClient {
...
+        const response = await this.httpClient.putAsync<
+            Contract,
+            Contract,
+            FusionApiHttpErrorResponse
+        >(url, contract);
+
+        return response.data;
+    }
+
+    async createExternalCompanyReprasentativeAsync(
```
Comment (olerichard) [inline src/frontend/src/api/ApiClient.ts]: it should be repre not repra. 
Representative

**Pull request merged**